ovs/proto_trace.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2017 DigitalOcean.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ovs
+
+import (
+	"encoding"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	// ErrInvalidProtoTrace is returned when the output from
+	// ovs-appctl ofproto/trace is in an unexpected format
+	ErrInvalidProtoTrace = errors.New("invalid ofproto/trace output")
+
+	datapathActionsRegexp = regexp.MustCompile(`Datapath actions: (.*)`)
```

Comment:
I'm _pretty_ sure these specific lines have been consistent, but we might want to sanity check versions before attempting to parse.  I know at least the individual trace of each table inside of the pipeline has changed, for instance.